server/src/main/java/org/opensearch/action/admin/cluster/remotestore/repository/RemoteStoreRepositoryRegistrationHelper.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.remotestore.repository;
+
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.cluster.metadata.RepositoriesMetadata;
+import org.opensearch.cluster.metadata.RepositoryMetadata;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.repositories.RepositoriesService;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * RemoteStore Repository Registration helper
+ */
+public class RemoteStoreRepositoryRegistrationHelper {
+
+    public static final String REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY = "remote_store.segment.repository.name";
+    public static final String REMOTE_STORE_SEGMENT_REPOSITORY_TYPE_ATTRIBUTE_KEY = "remote_store.segment.repository.type";
+    public static final String REMOTE_STORE_SEGMENT_REPOSITORY_SETTINGS_ATTRIBUTE_KEY = "remote_store.segment.repository.settings";
+    public static final String REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY = "remote_store.translog.repository.name";
+    public static final String REMOTE_STORE_TRANSLOG_REPOSITORY_TYPE_ATTRIBUTE_KEY = "remote_store.translog.repository.type";
+    public static final String REMOTE_STORE_TRANSLOG_REPOSITORY_SETTINGS_ATTRIBUTE_KEY = "remote_store.translog.repository.settings";
```

Comment:
we are taking all repo details twice. With remote metadata , this redundancy would increase to three . Can we just take the repo name as an input and not put all the details in node attributes ? For bigger clusters and bigger settings blobs, this can bloat up the cluster state as well.

Reply:
Also we will need to reconcile if same repo is used for all the stores.